go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -565,13 +565,8 @@ func (n *NodeController) ensureHybridOverlayBridge(node *kapi.Node) error {
 	n.drMAC = portMAC
 
 	// n.drIP is always 3rd address in the subnet
-	// TODO add support for ipv6 later
-	portIP := subnet.IP.To4()
-	if portIP == nil {
-		return fmt.Errorf("failed to parse local node subnet: %s", subnet.IP)
-	}
-	portIP[3] += 3
-	n.drIP = portIP
+	hybridOverlayIfAddr := util.GetNodeHybridOverlayIfAddr(subnet)
```

Comment:
thanks for catching the direct byte manipulation on the original subnet object. However, I don't really like the fix. Those util functions just recursively call each other do to NextIP 3 times. I would rather you fix those functions to do indirect byte arithmetic or some other more efficient method than calling 2 other functions to do recursively do x.NextIP().NextIP().NextIP()

Reply:
@trozet I'd honestly like to leave it as-is since I'm not touching that part of the code the byte manipulation was a regression. I'd view changes to the IfAddr() stuff as a separate enhancement.

Reply:
OK fair enough. Filed:
https://github.com/ovn-org/ovn-kubernetes/issues/1386